haproxy-spoe-auth-operator/src/haproxy_spoe_auth_service.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The haproxy-spoe-auth service module."""
+
+import logging
+from pathlib import Path
+
+from charmlibs import snap
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+from state import CharmState, OauthInformation
+
+SNAP_NAME = "haproxy-spoe-auth"
+CONFIG_PATH = Path("/var/snap/haproxy-spoe-auth/current/config.yaml")
+CONFIG_TEMPLATE = "config.yaml.j2"
+
+logger = logging.getLogger(__name__)
+
+
+class SpoeAuthServiceInstallError(Exception):
+    """Exception raised when snap installation fails."""
+
+
+class SpoeAuthServiceConfigError(Exception):
+    """Exception raised when service configuration fails."""
+
+
+class SpoeAuthService:
+    """HAProxy SPOE Auth service class."""
+
+    def __init__(self) -> None:
+        """Initialize the service."""
+        self._template_dir = Path(__file__).parent.parent / "templates"
+        cache = snap.SnapCache()
+        self.haproxy_spoe_auth_snap = cache[SNAP_NAME]
+
+    def install(self) -> None:
+        """Install the haproxy-spoe-auth snap.
+
+        Raises:
+            SpoeAuthServiceInstallError: When snap installation fails.
+        """
+        try:
+            if not self.haproxy_spoe_auth_snap.present:
+                self.haproxy_spoe_auth_snap.ensure(snap.SnapState.Latest, channel="edge")
```

Comment:
I'm confused, isn't it "latest"?
```suggestion
                self.haproxy_spoe_auth_snap.ensure(snap.SnapState.Latest, channel=SNAP_CHANNEL)
```